### PR TITLE
Standardize usage of string type comparison

### LINKED ISSUE #1823: Inconsistent usage of str, basestring, string_types
There's a lot of mix-and-match through Pyomo about whether to use `str`, `basestring`, or `string_types` when using `isinstance`. Since we've dropped Python 2 support, we should standardize all of these to `str`.

### PR DESCRIPTION
**Pull request by mrmundt**
## Summary/Motivation:
This addresses an inconsistency in usage of type comparison relating to string. Since we no longer support Python 2.x, we can convert to using just the Python 3.x standard.

## Changes proposed in this PR:
- Replace all instances of `basestring` and `six.string_types` with `str`

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/core/base/PyomoModel.py [hunk 1]
```diff
@@ -829,6 +825,8 @@ def _load_model_data(self, modeldata, namespaces, **kwds):
 
                 self._initialize_component(modeldata, namespaces, component_name, profile_memory)
                 if False:
+                    # !!THIS SEEMS LIKE A BUG!! - mrmundt
+                    # there is no start_time
```
Comment (blnicho) [inline pyomo/core/base/PyomoModel.py:829]: Agreed, this whole if-block is unreachable and likely there for reference. @jsiirola looks like you were the last one to touch these lines (3 years ago), can this if-block be removed?

File: pyomo/core/expr/numvalue.py [hunk 2]
```diff
@@ -99,12 +99,6 @@ def __setstate__(self, state):
 native_boolean_types = set([ int, bool, str ])
 native_logical_types = {bool, }
 pyomo_constant_types = set()  # includes NumericConstant
-try:
-    native_numeric_types.add(long)
-    native_integer_types.add(long)
-    native_boolean_types.add(long)
-except:
-    pass
```
Comment (blnicho) [inline pyomo/core/expr/numvalue.py:107]: Why is this no longer needed?

Comment (mrmundt) [reply pyomo/core/expr/numvalue.py:107]: `long` no longer exists in Python 3.x. It's a Python 2 integer type.

**Comment by mrmundt** [comment]
@blnicho , @jsiirola - I'm not sure if the timeout failure in OSX 3.9 is a real failure. Any insights?

File: pyomo/core/base/PyomoModel.py (hunk 1 above)
Comment (jsiirola) [reply pyomo/core/base/PyomoModel.py:829]: Yes.  I think thus can be removed.  My guess is that this was the old "construction time reporting" that was superseded by the new timing reporting in `pyomo.common.timing`.

**Review by blnicho — APPROVED** [review]

File: pyomo/util/slices.py [hunk 3]
```diff
@@ -13,10 +13,9 @@
 from pyomo.core.base.indexed_component import normalize_index
 from pyomo.core.base.indexed_component_slice import IndexedComponent_slice
 from pyomo.core.base.global_set import UnindexedComponent_set
-from pyomo.common.collections import ComponentSet, ComponentMap
 
 def _to_iterable(source):
-    iterable_scalars = six.string_types + (six.binary_type, six.text_type)
+    iterable_scalars = (str,) + (six.binary_type, six.text_type)
```
Comment (jsiirola) [inline pyomo/util/slices.py:18]: This should just be a single explicit tuple, `(str, bytes)`

**Review by jsiirola — APPROVED** [review]

**Pull request merged**